src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -4,21 +4,30 @@
 import logging
 import cherrypy
 
-from . import ApiController, RESTController
+from . import ApiController, RESTController, ControllerDoc, EndpointDoc
 from .. import mgr
 from ..exceptions import DashboardException
 from ..services.auth import AuthManager, JwtManager
 
 
 logger = logging.getLogger('controllers.auth')
 
+AUTH_CHECK_SCHEMA = {
+    "username": (str, "Username"),
+    "permissions": ({
+        "cephfs": ([str], "")
+    }, "List of permiisions acquired"),
+    "sso": (bool, ""),
+    "pwdUpdateRequired": (bool, "Is password update required?")
+}
```

Comment:
Just an improvement suggestion (perhaps not for this PR, as it's an awesome job).

It looks like this schema and the ones in `qa/tasks/mgr/dashboard/...` are more or less the same kind of struct:
```py
         {                                                     
             'token': JLeaf(str),                                                                     
             'username': JLeaf(str),                                                                  
             'permissions': JObj(sub_elems={}, allow_unknown=True),                                   
             'sso': JLeaf(bool),                                                                      
             'pwdExpirationDate': JLeaf(int, none=True),                                              
             'pwdUpdateRequired': JLeaf(bool)                                                                                             
         }
```

Could we share/re-use this schemas across OpenAPI stuff and back-end API tests?

Python 3.8 `typing` introduced [`TypedDict`](https://docs.python.org/3/library/typing.html#typing.TypedDict) ([PEP-589](https://www.python.org/dev/peps/pep-0589/)):

```py
class AuthCheckResponse(TypedDict):
    username: str
    permissions: Permissions
    sso: bool
    pwdUpdateRequired: bool
```

That only supports types, not descriptions or other metadata supported by OpenAPI, so we could simply create wrapping type (e.g.: `APIType`):
```py
class AuthCheckResponse(TypedDict):
    username: APIType(str, "Username")
    permissions: APIType(Permissions, "List of permissions acquired")
    sso: bool
    pwdUpdateRequired: APIType(bool, "Is password update required?")
}
```

TypeDict is not a real class, but a dictionary wrapper, so any check here will happen statically (mypy). If we want to do some runtime checks, we would need something else. There's a Python library 'attrs' with a [very good package support](https://pkgs.org/download/python3-attrs), that provides [this functionality](https://www.attrs.org/en/stable/types.html).

Reply:
Thanks @epuertat.! In my opinion, the OpenAPI would not support the **JLeaf** and **Jobj** types, anyways will try to reuse the schemas if possible and learn about the TypeDict wrapper to improve this functionality.

Reply:
@epuertat @aaSharma14 Regarding ```TypedDict``` AFAIK master is used Python 3.6 (also in API tests job), not Python 3.8.

Reply:
> Thanks @epuertat.! In my opinion, the OpenAPI would not support the **JLeaf** and **Jobj** types, anyways will try to reuse the schemas if possible and learn about the TypeDict wrapper to improve this functionality.

Yes, I don't mean that, I'm just thinking on how to reuse this across documentation, static, and runtime checks (which is something we don't have yet).